params.go

```diff
@@ -338,55 +341,30 @@ func (p *Params) SetDefaults(appLabel, buildVersion, releaseName, releaseAction
 		p.Container.Lifecycle.PrestopSleepSeconds = &defaultSleepValue
 	}
 
-	// set sidecar defaults
-	if p.Sidecar.Type == "" {
-		switch p.Kind {
-		case "job":
-			p.Sidecar.Type = "none"
+	// Code for backwards-compatibility: in the parameters the sidecar can be specified both in the "sidecar" field, and also as an element in the "sidecars" collection.
+	// The "sidecar" field is kept around for backwards compatibility, but due to this we need some extra checks to cover all cases.
+	openrestySidecarSpecified := p.Sidecar.Type == "openresty"
```

Comment:
Perhaps rename this variable to `deprecatedOpenrestySidecarSpecified` or `legacyOpenrestySidecarSpecified` to make it a bit more clear that it's the old way of doing stuff. We could also issue a warning in the logs that it can be replaced with the new `sidecars:` array.

Reply:
Is there an established way to output a warning? I could do this check in the `ValidateRequiredProperties` in `params.go`, and do a `log.Printf` there. Is there a better way?

Reply:
Nope, not yet; I think most of it happens in main.go, but it wouldn't be nice to put it there. Unless besides the validation errors we also returning warnings, to be shown but not halt execution.

Reply:
I added a third return value, a list of strings for the warnings.  
It shows up in the logs like this (example: https://estafette.travix.com/pipelines/github.com/estafette/estafette-extension-gke/builds/434484848076488709/logs)
```
2019-03-15 16:45:15
Warning: The sidecar field is deprecated, the sidecars list should be used instead.
```

Reply:
Cool stuff